// @fluentui/react-drawer/lib-commonjs/components/DrawerOverlay/useDrawerOverlay.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useDrawerOverlay_unstable", {
    enumerable: true,
    get: function() {
        return useDrawerOverlay_unstable;
    }
});
const _interop_require_wildcard = require("@swc/helpers/_/_interop_require_wildcard");
const _react = /*#__PURE__*/ _interop_require_wildcard._(require("react"));
const _reactutilities = require("@fluentui/react-utilities");
const _reactdialog = require("@fluentui/react-dialog");
const _reactmotionpreview = require("@fluentui/react-motion-preview");
const _useDrawerDefaultProps = require("../../shared/useDrawerDefaultProps");
const _DrawerOverlaySurface = require("./DrawerOverlaySurface");
const useDrawerOverlay_unstable = (props, ref)=>{
    const { open, size, position } = (0, _useDrawerDefaultProps.useDrawerDefaultProps)(props);
    const { modalType = 'modal', inertTrapFocus, defaultOpen = false, onOpenChange } = props;
    const drawerMotion = (0, _reactmotionpreview.useMotion)(open);
    const backdropMotion = (0, _reactmotionpreview.useMotion)(open);
    const backdropInnerProps = _reactutilities.slot.resolveShorthand(props.backdrop);
    const hasCustomBackdrop = modalType !== 'non-modal' && backdropInnerProps !== null;
    const backdropProps = {
        ...backdropInnerProps,
        ref: (0, _reactutilities.useMergedRefs)(backdropMotion.ref, backdropInnerProps === null || backdropInnerProps === void 0 ? void 0 : backdropInnerProps.ref)
    };
    const root = _reactutilities.slot.always({
        ...props,
        backdrop: hasCustomBackdrop ? backdropProps : null
    }, {
        elementType: _DrawerOverlaySurface.DrawerOverlaySurface,
        defaultProps: {
            ref: (0, _reactutilities.useMergedRefs)(ref, drawerMotion.ref)
        }
    });
    const dialog = _reactutilities.slot.always({
        open: true,
        defaultOpen,
        onOpenChange,
        inertTrapFocus,
        modalType,
        /*
       * children is not needed here because we construct the children in the render function,
       * but it's required by DialogProps
       */ children: null
    }, {
        elementType: _reactdialog.Dialog
    });
    return {
        components: {
            root: _DrawerOverlaySurface.DrawerOverlaySurface,
            dialog: _reactdialog.Dialog
        },
        root,
        dialog,
        size,
        position,
        motion: drawerMotion,
        backdropMotion
    };
};
