// @fluentui/react-combobox/lib-commonjs/utils/useComboboxPopup.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useComboboxPopup", {
    enumerable: true,
    get: function() {
        return useComboboxPopup;
    }
});
const _reactpositioning = require("@fluentui/react-positioning");
const _reactutilities = require("@fluentui/react-utilities");
function useComboboxPopup(props, triggerShorthand, listboxShorthand) {
    const { positioning } = props;
    // Set a default set of fallback positions to try if the dropdown does not fit on screen
    const fallbackPositions = [
        'above',
        'after',
        'after-top',
        'before',
        'before-top'
    ];
    // popper options
    const popperOptions = {
        position: 'below',
        align: 'start',
        offset: {
            crossAxis: 0,
            mainAxis: 2
        },
        fallbackPositions,
        ...(0, _reactpositioning.resolvePositioningShorthand)(positioning)
    };
    const { targetRef, containerRef } = (0, _reactpositioning.usePositioning)(popperOptions);
    const listboxRef = (0, _reactutilities.useMergedRefs)(listboxShorthand === null || listboxShorthand === void 0 ? void 0 : listboxShorthand.ref, containerRef);
    const listbox = listboxShorthand && {
        ...listboxShorthand,
        ref: listboxRef
    };
    return [
        {
            ...triggerShorthand,
            ref: (0, _reactutilities.useMergedRefs)(triggerShorthand === null || triggerShorthand === void 0 ? void 0 : triggerShorthand.ref, targetRef)
        },
        listbox
    ];
}
