// @fluentui/react-drawer/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Drawer: function() {
        return _Drawer.Drawer;
    },
    renderDrawer_unstable: function() {
        return _Drawer.renderDrawer_unstable;
    },
    useDrawer_unstable: function() {
        return _Drawer.useDrawer_unstable;
    },
    DrawerOverlay: function() {
        return _DrawerOverlay.DrawerOverlay;
    },
    drawerOverlayClassNames: function() {
        return _DrawerOverlay.drawerOverlayClassNames;
    },
    renderDrawerOverlay_unstable: function() {
        return _DrawerOverlay.renderDrawerOverlay_unstable;
    },
    useDrawerOverlayStyles_unstable: function() {
        return _DrawerOverlay.useDrawerOverlayStyles_unstable;
    },
    useDrawerOverlay_unstable: function() {
        return _DrawerOverlay.useDrawerOverlay_unstable;
    },
    DrawerInline: function() {
        return _DrawerInline.DrawerInline;
    },
    drawerInlineClassNames: function() {
        return _DrawerInline.drawerInlineClassNames;
    },
    renderDrawerInline_unstable: function() {
        return _DrawerInline.renderDrawerInline_unstable;
    },
    useDrawerInlineStyles_unstable: function() {
        return _DrawerInline.useDrawerInlineStyles_unstable;
    },
    useDrawerInline_unstable: function() {
        return _DrawerInline.useDrawerInline_unstable;
    },
    DrawerBody: function() {
        return _DrawerBody.DrawerBody;
    },
    drawerBodyClassNames: function() {
        return _DrawerBody.drawerBodyClassNames;
    },
    renderDrawerBody_unstable: function() {
        return _DrawerBody.renderDrawerBody_unstable;
    },
    useDrawerBodyStyles_unstable: function() {
        return _DrawerBody.useDrawerBodyStyles_unstable;
    },
    useDrawerBody_unstable: function() {
        return _DrawerBody.useDrawerBody_unstable;
    },
    DrawerHeader: function() {
        return _DrawerHeader.DrawerHeader;
    },
    drawerHeaderClassNames: function() {
        return _DrawerHeader.drawerHeaderClassNames;
    },
    renderDrawerHeader_unstable: function() {
        return _DrawerHeader.renderDrawerHeader_unstable;
    },
    useDrawerHeaderStyles_unstable: function() {
        return _DrawerHeader.useDrawerHeaderStyles_unstable;
    },
    useDrawerHeader_unstable: function() {
        return _DrawerHeader.useDrawerHeader_unstable;
    },
    DrawerHeaderTitle: function() {
        return _DrawerHeaderTitle.DrawerHeaderTitle;
    },
    drawerHeaderTitleClassNames: function() {
        return _DrawerHeaderTitle.drawerHeaderTitleClassNames;
    },
    renderDrawerHeaderTitle_unstable: function() {
        return _DrawerHeaderTitle.renderDrawerHeaderTitle_unstable;
    },
    useDrawerHeaderTitleStyles_unstable: function() {
        return _DrawerHeaderTitle.useDrawerHeaderTitleStyles_unstable;
    },
    useDrawerHeaderTitle_unstable: function() {
        return _DrawerHeaderTitle.useDrawerHeaderTitle_unstable;
    },
    DrawerHeaderNavigation: function() {
        return _DrawerHeaderNavigation.DrawerHeaderNavigation;
    },
    drawerHeaderNavigationClassNames: function() {
        return _DrawerHeaderNavigation.drawerHeaderNavigationClassNames;
    },
    renderDrawerHeaderNavigation_unstable: function() {
        return _DrawerHeaderNavigation.renderDrawerHeaderNavigation_unstable;
    },
    useDrawerHeaderNavigationStyles_unstable: function() {
        return _DrawerHeaderNavigation.useDrawerHeaderNavigationStyles_unstable;
    },
    useDrawerHeaderNavigation_unstable: function() {
        return _DrawerHeaderNavigation.useDrawerHeaderNavigation_unstable;
    },
    DrawerFooter: function() {
        return _DrawerFooter.DrawerFooter;
    },
    drawerFooterClassNames: function() {
        return _DrawerFooter.drawerFooterClassNames;
    },
    renderDrawerFooter_unstable: function() {
        return _DrawerFooter.renderDrawerFooter_unstable;
    },
    useDrawerFooterStyles_unstable: function() {
        return _DrawerFooter.useDrawerFooterStyles_unstable;
    },
    useDrawerFooter_unstable: function() {
        return _DrawerFooter.useDrawerFooter_unstable;
    }
});
const _Drawer = require("./Drawer");
const _DrawerOverlay = require("./DrawerOverlay");
const _DrawerInline = require("./DrawerInline");
const _DrawerBody = require("./DrawerBody");
const _DrawerHeader = require("./DrawerHeader");
const _DrawerHeaderTitle = require("./DrawerHeaderTitle");
const _DrawerHeaderNavigation = require("./DrawerHeaderNavigation");
const _DrawerFooter = require("./DrawerFooter");
